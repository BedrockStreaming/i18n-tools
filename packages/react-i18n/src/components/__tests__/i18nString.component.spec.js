import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { Trans } from '../i18nString.component';

describe('i18n.renderProps', () => {
  let context = {};
  const childContextTypes = {
    getTranslateFunction: PropTypes.func,
  };

  beforeEach(() => {
    context = {
      getTranslateFunction: jest.fn(() => jest.fn(x => x)),
    };
  });

  it('should return translation string', () => {
    const wrapper = shallow(<Trans i18nKey="foo.bar" />, { context, childContextTypes });

    expect(wrapper).toMatchSnapshot();
  });
});
