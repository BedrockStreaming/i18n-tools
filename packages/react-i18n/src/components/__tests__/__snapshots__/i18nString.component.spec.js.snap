// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n.renderProps should return translation string 1`] = `"foo.bar"`;
